examples/go/tagging/handler/handler.go

```diff
@@ -31,140 +32,145 @@ type vcConfig struct {
+		err := vsConnect()
+		if err != nil {
+			log.Fatalf("connect to vSphere failed: %v", err)
+			return handler.Response{
+				Body:       nil,
+				StatusCode: http.StatusBadRequest,
+			}, fmt.Errorf("connect to vSphere failed: %w", err)
+		}
 	}
 
-	// Incoming request object.
-	var event vmEvent
-	err := json.Unmarshal(req.Body, &event)
+	// Set up os signal handling to log out of vSphere.
+	signal.Notify(sigCh, syscall.SIGTERM, os.Interrupt)
+	go handleSignal(ctx, sigCh)
+
+	// Retrieve the Managed Object Reference from the event.
+	moRef, err := parseEventMoRef(req.Body)
 	if err != nil {
+		log.Fatalf("retrieve managed reference object failed: %v", err)
 		return handler.Response{
 			Body:       nil,
 			StatusCode: http.StatusBadRequest,
 		}, err
 	}
 
-	if event.Event.Vm.Vm.Value == "" || event.Event.Vm.Vm.Type == "" {
+	log.Println("tag VM moRef with tagID (URN)")
+	err = client.MoTag(ctx, moRef, cfg.Tag.Urn)
+	if err != nil {
+		log.Fatalf("tagging managed reference object failed: %v", err)
 		return handler.Response{
 			Body:       nil,
 			StatusCode: http.StatusBadRequest,
-		}, errors.New("managed Reference Object information must not be empty")
+		}, err
 	}
 
-	// Fill information in the request into a govmomi type.
-	var moRef types.ManagedObjectReference
-	moRef.Value = event.Event.Vm.Vm.Value
-	moRef.Type = event.Event.Vm.Vm.Type
-
-	// Connect to vcenter with govmomi client
-	log.Println("connect to vCenter")
-	clients := connect(ctx, vcc.VCenter)
-	log.Println("connection successful")
-
-	// tag the VM moRef with tagID (URN)
-	clients.tagVM(ctx, moRef, vcc.Tag.Urn)
-
 	// Respond with a message that tagging of vm was complete
 	message := fmt.Sprintf(
 		"%v was tagged with %v",
 		moRef.Value,
-		vcc.Tag.Urn,
+		cfg.Tag.Urn,
 	)
 	log.Println(message)
 
-	err = clients.Govmomi.Logout(ctx)
-	if err != nil {
-		log.Printf("unable to log out of govmomi due to %v", err)
-	}
-	err = clients.Rest.Logout(ctx)
-	if err != nil {
-		log.Printf("unable to log out of rest due to %v", err)
-	}
-
 	return handler.Response{
 		Body:       []byte(message),
 		StatusCode: http.StatusOK,
 	}, nil
 }
 
-func loadTomlCfg() (vcConfig, error) {
-	secretPath := "/var/openfaas/secrets/vcconfig"
-	secret, err := toml.LoadFile(secretPath)
+// vsConnect connects to vSphere govmomi API using information from vcconfig.toml.
+func vsConnect() error {
+	var (
+		err        error
+		configPath = "/var/openfaas/secrets/vcconfig"
+	)
+
+	// Retrieve vCenter and tag info from the vcconfig file.
+	log.Println("load vcconfig.toml")
+	cfg, err = loadTomlCfg(configPath)
 	if err != nil {
-		return nil, err
+		return fmt.Errorf("vcconfig.toml loading failed: %w", err)
 	}
 
-	cfg := vcConfig{}
-	secret.Unmarshal(&cfg)
+	u := url.URL{
+		Scheme: "https",
+		Host:   cfg.VCenter.Server,
+		Path:   "sdk",
+	}
+	u.User = url.UserPassword(cfg.VCenter.User, cfg.VCenter.Password)
 
-	return cfg, nil
-}
+	// Currently, VEBA does not support signed certificates.
```

Comment:
This is not correct. Your fn decides whether it will support self-signed certs. Consider making this an ENV var so users can opt in/out.